src/os/ObjectStore.h

```diff
@@ -992,6 +994,19 @@ class ObjectStore {
       _op->op = OP_NOP;
       data.ops++;
     }
+    /**
+     * create
+     *
+     * create an object that does not yet exist
+     * (behavior is undefined if the object already exists)
```

Comment:
Perhaps it's better to check for onode presence and  assert if it exists. This is a bit less effective from performance point of view but IMO troubleshooting after 'undefined behavior' is much worse.

Reply:
I agree it's hard to troubleshoot, but the whole point of this PR is to trade that for performance.  This is why I think we should merge this after nautilus so we have a long cycle to shake out problems.